lib/dor/models/permissable.rb

```diff
@@ -0,0 +1,160 @@
+module Dor
+  module Permissable
+    extend ActiveSupport::Concern
+
+    # General documentation about roles and permissions is on SUL Consul at
+    # https://consul.stanford.edu/display/chimera/Repository+Roles+and+Permissions
+    # All these constants are frozen arrays so the methods that use them can
+    # easily add them to return arrays.
+    SDR_ADMINS = %w(sdr-administrator).freeze
+    SDR_MANAGERS = %w(sdr-manager).freeze
+    SDR_VIEWERS = %w(sdr-viewer).freeze
+
+    APO_MANAGERS = %w(dor-apo-manager).freeze
+    APO_DEPOSITORS = %w(dor-apo-depositor).freeze
+    APO_METADATA = %w(dor-apo-metadata).freeze
+    APO_VIEWERS = %w(dor-apo-viewer).freeze
+
+    # A complete set of known roles.  This can be used by clients to
+    # inspect all the possible roles available.
+    KNOWN_ROLES = (
+      SDR_ADMINS + SDR_MANAGERS + SDR_VIEWERS +
+      APO_MANAGERS + APO_DEPOSITORS + APO_METADATA + APO_VIEWERS
+    ).freeze
+
+    # ---
+    # APO permissions
+
+    def can_create_apo?(roles)
+      intersect roles, roles_which_create_apo
+    end
+
+    def can_manage_apo?(roles)
+      intersect roles, roles_which_manage_apo
+    end
+
+    def can_manage_collections?(roles)
+      intersect roles, roles_which_manage_collections
+    end
+
+    def can_manage_roles?(roles)
+      intersect roles, roles_which_manage_roles
+    end
+
+    def can_manage_sets?(roles)
+      intersect roles, roles_which_manage_sets
+    end
+
+    def can_release_objects?(roles)
+      intersect roles, roles_which_release_objects
+    end
+
+    # ---
+    # Item permissions
+
+    def can_manage_item?(roles)
+      intersect roles, roles_which_manage_item
+    end
+
+    def can_register_item?(roles)
+      intersect roles, roles_which_register_item
+    end
+
+    def can_manage_desc_metadata?(roles)
+      intersect roles, roles_which_manage_desc_md
+    end
+
+    def can_manage_system_metadata?(roles)
+      intersect roles, roles_which_manage_sys_md
+    end
+
+    def can_manage_contents?(roles)
+      intersect roles, roles_which_manage_contents
+    end
+
+    def can_manage_rights?(roles)
+      intersect roles, roles_which_manage_rights
+    end
+
+    def can_manage_workflows?(roles)
+      intersect roles, roles_which_manage_workflows
+    end
+
+    def can_manage_embargo?(roles)
+      intersect roles, roles_which_manage_embargo
+    end
+
+    # ---
+    # Common viewing permissions
+
+    def can_view_content?(roles)
+      intersect roles, roles_which_view_content
+    end
+
+    def can_view_metadata?(roles)
+      intersect roles, roles_which_view_metadata
+    end
```

Comment:
The `private` methods can be modified to change role permissions without changing the public `can_*` methods.